src/titiler/xarray/titiler/xarray/io.py

```diff
@@ -0,0 +1,306 @@
+    ds = xarray.open_dataset(file_handler, **xr_open_args)
+
+    if cache_client:
+        # Serialize the dataset to bytes using pickle
+        data_bytes = pickle.dumps(ds)
+        cache_client.set(cache_key, data_bytes)
+
+    return ds
+
+
+def _arrange_dims(da: xarray.DataArray) -> xarray.DataArray:
+    """Arrange coordinates and time dimensions.
+
+    An rioxarray.exceptions.InvalidDimensionOrder error is raised if the coordinates are not in the correct order time, y, and x.
+    See: https://github.com/corteva/rioxarray/discussions/674
+
+    We conform to using x and y as the spatial dimension names..
+
+    """
+    if "x" not in da.dims and "y" not in da.dims:
+        try:
+            latitude_var_name = next(
+                x for x in ["lat", "latitude", "LAT", "LATITUDE", "Lat"] if x in da.dims
+            )
+            longitude_var_name = next(
+                x
+                for x in ["lon", "longitude", "LON", "LONGITUDE", "Lon"]
+                if x in da.dims
+            )
+        except StopIteration as e:
+            raise ValueError(f"Couldn't find X/Y dimensions in {da.dims}") from e
+
+        da = da.rename({latitude_var_name: "y", longitude_var_name: "x"})
+
+    if "TIME" in da.dims:
+        da = da.rename({"TIME": "time"})
+
+    if _dims := [d for d in da.dims if d not in ["x", "y"]]:
+        da = da.transpose(*_dims, "y", "x")
+    else:
+        da = da.transpose("y", "x")
+
+    # If min/max values are stored in `valid_range` we add them in `valid_min/valid_max`
+    vmin, vmax = da.attrs.get("valid_min"), da.attrs.get("valid_max")
+    if "valid_range" in da.attrs and not (vmin is not None and vmax is not None):
+        valid_range = da.attrs.get("valid_range")
+        da.attrs.update({"valid_min": valid_range[0], "valid_max": valid_range[1]})
+
+    return da
+
+
+def get_variable(
+    ds: xarray.Dataset,
+    variable: str,
+    datetime: Optional[str] = None,
+    drop_dim: Optional[str] = None,
+) -> xarray.DataArray:
+    """Get Xarray variable as DataArray.
+
+    Args:
+        ds (xarray.Dataset): Xarray Dataset.
+        variable (str): Variable to extract from the Dataset.
+        datetime (str, optional): datetime to select from the DataArray.
+        drop_dim (str, optional): DataArray dimension to drop in form of `{dimension}={value}`.
+
+    Returns:
+        xarray.DataArray: 2D or 3D DataArray.
+
+    """
+    da = ds[variable]
+
+    if drop_dim:
+        dim_to_drop, dim_val = drop_dim.split("=")
+        da = da.sel({dim_to_drop: dim_val}).drop_vars(dim_to_drop)
+
+    da = _arrange_dims(da)
+
+    # Make sure we have a valid CRS
+    crs = da.rio.crs or "epsg:4326"
+    da = da.rio.write_crs(crs)
+
+    if crs == "epsg:4326" and (da.x > 180).any():
+        # Adjust the longitude coordinates to the -180 to 180 range
+        da = da.assign_coords(x=(da.x + 180) % 360 - 180)
+
+        # Sort the dataset by the updated longitude coordinates
+        da = da.sortby(da.x)
```

Comment:
I wonder if there are more CRS definitions we would want to apply this fix to. Maybe there is a way to tell if we want to adjust coordinates based on some other CRS properties besides an exact name match.

Reply:
rasterio doesn't have a `CRS.is_geographic` method yet (https://github.com/rasterio/rasterio/pull/3218) but once it's available we could check if the CRS is geographic and then run those fixes